+nosnoc/+solver/MpccSolver.m

```diff
@@ -426,5 +458,415 @@ function print_infeasibility(obj, results)
                 keyboard
             end
         end
+
+        function [polished_w, res_out, stat_type, n_biactive] = calculate_stationarity(obj, exitfast, lifted)
+            import casadi.*
+            stat_type = "?";
+            nlp = obj.nlp;
+            mpcc = obj.mpcc;
+
+            % This method generates a TNLP (tightened NLP) using the results of a homotopy solver.
+            % In order to do so we first identify the active set of the complementarity constraints which involves identifying which
+            % branch of the complementarity each element of G(w) and H(w) are, or whether the solution is biactive at that point.
+            % For a more in depth explanation see the PhD theses of Armin Nurkanovic or Alexandra Schwartz
+            
+            w_orig = nlp.w.mpcc_w().res;
+            lam_x = nlp.w.mpcc_w().mult;
+            lam_g = nlp.g.mpcc_g().mult;
+
+            G = mpcc.G;
+            H = mpcc.H;
+            G_old = full(obj.G_fun(w_orig, nlp.p.mpcc_p().val));
+            H_old = full(obj.H_fun(w_orig, nlp.p.mpcc_p().val));
+            if obj.opts.homotopy_steering_strategy == HomotopySteeringStrategy.DIRECT
+                a_tol = 1*sqrt(obj.opts.comp_tol);
```

Comment:
add comment why sqrt and why 1*

Reply:
Because we generally have `a*b-sigma` (at least for Scholtes). In this case the intersection of the level set at sigma and a=b is `sqrt(sigma)`. Added a comment.